runtime/jcl/common/jclcinit.c

```diff
@@ -712,6 +717,35 @@ initializeRequiredClasses(J9VMThread *vmThread, char* dllName)
 		}
 	}
 
+	if (J9_ARE_ANY_BITS_SET(vm->extendedRuntimeFlags2, J9_EXTENDED_RUNTIME2_LOAD_AGENT_MODULE)
+			&& (NULL != vm->modulesPathEntry->extraInfo)) {
+		Trc_JCL_initializeRequiredClasses_addAgentModuleEntry(vmThread);
+		const char *module = vm->jimageIntf->jimagePackageToModule(
+				vm->jimageIntf, (UDATA) vm->modulesPathEntry->extraInfo,
+				"jdk/internal/agent");
+		if (NULL != module) {
+			J9VMSystemProperty *systemProperty = NULL;
+			/* Handle the case where there are no user-specified modules. In this case, there
+			 * is typically one system property but no user-set "add-modules" arguments.
+			 */
+			if ((0 == vm->addModulesCount) &&
+					(J9SYSPROP_ERROR_NONE == vm->internalVMFunctions->getSystemProperty(vm, ADDMODS_PROPERTY_BASE "0", &systemProperty))
+					) {
+				/* this is implicitly an unused property */
+				vm->internalVMFunctions->setSystemProperty(vm, systemProperty, AGENT_MODULE_NAME);
+			} else {
+				UDATA indexLen = (UDATA)floor(log10((double)vm->addModulesCount) + 1);
```

Comment:
This is computing the printed length of the value, which I think could be better done with a str_printf on NULL. Also fix vmprops.c at the same time.

Reply:
I suppose the math might be a little faster for startup, but it's a lot more confusing.

@DanHeidinga What do you think?

Reply:
@gacholio I like your solution.
Updated the length calculation in both places.

Reply:
It may be worth turning this into a macro/helper with a descriptive name.

Reply:
> which I think could be better done with a str_printf on NULL.

 +1 to this from me.